### PR TITLE
Add current_repodata_hack

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by isuruf** [comment]
cc @beckermr, @leofang

**Review by beckermr — CHANGES_REQUESTED** [review]
I think we should prepend the outputs with _current_repodata_hack so that it is clear they are not related to the actual compilers.

File: recipes/_current_repodata_hack/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  - name: _current_repodata_hack_gcc_linux_ppc64le_84
+    build:
+      noarch: generic
+    requirements:
+      - gcc_linux-ppc64le 8.4.*
+
+about:
+  home: https://github.com/conda-forge/_current_repodata_hack-feedstock
+  license: LicenseRef-OTHER
+  summary: Meta-package to fix current_repodata
```
Comment (beckermr) [inline recipes/_current_repodata_hack/meta.yaml:53]:
```suggestion
  summary: Meta-package to fix current_repodata
  description: |
    As of writing, conda-forge has versions of the linux compilers that are ahead of the 
    ones in the global pinnings. The current_repodata.json file only pulls in the latest 
    version of a package by default. Thus anytime one asks for the compilers in the pinnings,
    the solver will always fail on current_repodata.json and have to pull all of the repodata. 
    The packages here make sure the latest versions of the compilers we use are in 
    current_repodata.json by depending on those packages. Thus it solves the solver inefficiency. 
    It also helps in cases when the solver appears to pull older builds of the compilers when they are 
    not explicitly asked for in the environment.
```

**Review by beckermr — COMMENTED** [review]
Thank you for changing the names. I added one more comment with some extra text explaining why these packages exist. This will help me not forget!

**Review by beckermr — APPROVED** [review]

**Comment by leofang** [comment]
Thanks! Looks like it's fixed: https://github.com/conda-forge/openmpi-feedstock/issues/61#issuecomment-695081644.

**Pull request merged**